org.abapgit.adt.ui/src/org/abapgit/adt/ui/internal/wizards/AbapGitWizardPageObjectsSelectionForPull.java

```diff
@@ -0,0 +1,225 @@
+
+		this.container = new Composite(parent, SWT.NONE);
+
+		GridLayout layout = new GridLayout();
+		this.container.setLayout(layout);
+
+		this.container.setLayoutData(new GridData(SWT.FILL, SWT.FILL, true, true));
+
+		this.modifiedObjTreeViewer = new CheckboxTreeViewer(this.container,
+				SWT.MULTI | SWT.BORDER | SWT.H_SCROLL | SWT.V_SCROLL | SWT.FILL);
+
+		this.modifiedObjTreeViewer.setContentProvider(new ModifiedObjectTreeContentProvider());
+
+		this.modifiedObjTreeViewer.getTree().setFocus();
+		this.modifiedObjTreeViewer.getTree().setHeaderVisible(true);
+		this.modifiedObjTreeViewer.getTree().setLinesVisible(true);
+		this.modifiedObjTreeViewer.getTree().setLayout(layout);
+		this.modifiedObjTreeViewer.getTree().setLayoutData(this.container.getLayoutData());
+
+		createColumns();
+
+		this.modifiedObjTreeViewer.setInput(this.repoToModifiedObjects);
+
+		setControl(this.container);
+		setPageComplete(true);
+
+		// Check the sub tree if parent is checked
+		this.modifiedObjTreeViewer.addCheckStateListener(new ICheckStateListener() {
+
+			@Override
+			public void checkStateChanged(CheckStateChangedEvent event) {
+				if (event.getElement() instanceof ModifiedObjectsForRepository) {
+					if (event.getChecked()) {
+						AbapGitWizardPageObjectsSelectionForPull.this.modifiedObjTreeViewer.setSubtreeChecked(event.getElement(), true);
+
+					} else {
+						AbapGitWizardPageObjectsSelectionForPull.this.modifiedObjTreeViewer.setSubtreeChecked(event.getElement(), false);
+					}
+				}
+
+			}
+		});
+
+	}
+
+	private void createColumns() {
+
+		// TYPE/OBJECT COLUMN
+		createTableViewerColumn("Name", 400).setLabelProvider(new ColumnLabelProvider() { //$NON-NLS-1$
+			@Override
+			public String getText(Object element) {
+				String result = ""; //$NON-NLS-1$
+
+				if (element instanceof ModifiedObjectsForRepository) {
+					result = ((ModifiedObjectsForRepository) element).getRepositoryURL();
+
+				} else if (element instanceof IAbapGitObject) {
+					result = ((IAbapGitObject) element).getName();
+				}
+				return result;
+			}
+
+		});
+
+
+		createTableViewerColumn("Package", 200).setLabelProvider(new ColumnLabelProvider() { //$NON-NLS-1$
+			@Override
+			public String getText(Object element) {
+				if (element instanceof IAbapGitObject) {
+					return ((IAbapGitObject) element).getPackageName();
+				}
+				return ""; //$NON-NLS-1$
+
+			}
+
+		});
+
+		createTableViewerColumn("Type", 100).setLabelProvider(new ColumnLabelProvider() { //$NON-NLS-1$
+			@Override
+			public String getText(Object element) {
+				if (element instanceof IAbapGitObject) {
+					return ((IAbapGitObject) element).getType();
+				}
+				return ""; //$NON-NLS-1$
+			}
+		});
+
+	}
+
+	private TreeViewerColumn createTableViewerColumn(String title, int bound) {
+		TreeViewerColumn viewerColumn = new TreeViewerColumn(this.modifiedObjTreeViewer, SWT.NONE);
+		viewerColumn.getColumn().setText(title);
+		viewerColumn.getColumn().setWidth(bound);
```

Comment:
We are using a `CheckBoxTreeViewer` for showing the modifiedObjects. Since, the underlying object is a Tree, we can't use `TableColumnLayout`.
So,  `TreeColumnLayout` is used instead.

Reply:
Cool